server/models/UserProfile.js

```diff
@@ -0,0 +1,47 @@
+const mongoose = require("mongoose");
+const User = require("./User");
+
+const Schema = mongoose.Schema,
+  ObjectId = Schema.ObjectId;
```

Comment:
nit: rest of the project uses the following style

```js
const Schema = mongoose.Schema;
const ObjectId = Schema.ObjectId;
```

I would use that instead of what you have now to be consistent to the overall project.